### PR TITLE
adds FAQ for null safety

### PR DESCRIPTION
**Pull request by ryanmcginnis**
preview: https://rm-dart.web.app/null-safety/faq

### CODE REVIEW

File: src/null-safety/faq.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
+---
+title: Null safety frequently asked questions
+description: FAQs for Dart's null safety migration
+toc: true
```
Comment (kwalrath) [inline src/null-safety/faq.md:4]: A shorter tab title seems wise here.

```suggestion
short-title: FAQ (null safety)
```

File: src/null-safety/faq.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
+---
+title: Null safety frequently asked questions
+description: FAQs for Dart's null safety migration
```
Comment (kwalrath) [inline src/null-safety/faq.md:3]: This is the page's social sharing text, so let's make it friendly and "you" focused.

```suggestion
description: FAQs to help you migrate your Dart code to null safety
```

File: src/null-safety/faq.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
+---
+title: Null safety frequently asked questions
```
Comment (kwalrath) [inline src/null-safety/faq.md:2]: Maybe a colon might help readability?

```suggestion
title: Null safety: frequently asked questions
```

File: src/null-safety/faq.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+description: FAQs for Dart's null safety migration
+toc: true
+---
+
```
Comment (kwalrath) [inline src/null-safety/faq.md:6]: We should add a brief intro here, with a link to /null-safety (at least).

We might also want to mention that this is based on an internal doc that's based on the experience of migrating Google internal code. (I'm not sure about this, but it makes me feel better about not enforcing our editorial style. :)

**Review by kwalrath — COMMENTED** [review]
I have some opinions (shocker! :) about the top matter and intro.

I'd usually do a copy edit pass over the rest of the page, but since this is a transitional doc that's based on an internal doc, I think it's OK not to match the official Google style.

File: src/null-safety/faq.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+An explicit runtime null check, for example `if (arg == null) throw
+ArgumentError(...)`, will be flagged as an unnecessary comparison if you make
+`arg` non-nullable.
+
+But, the check *is* still needed when running in mixed mode. Until everything is
+fully migrated and the code switches to running in sound mode, it will be
+possible for `arg` to be null.
+
+The simplest way to preserve behavior is change the check into
+[`ArgumentError.checkNotNull`](https://api.dart.dev/stable/2.9.3/dart-core/ArgumentError/checkNotNull.html).
```
Comment (kwalrath) [inline src/null-safety/faq.md:180]: In general, you should delete version numbers from URLs, so they automatically get the latest stable (or whatever) channel version of the doc.

```suggestion
[`ArgumentError.checkNotNull`](https://api.dart.dev/stable/dart-core/ArgumentError/checkNotNull.html).
```

File: src/null-safety/faq.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+```
+
+The default list constructor fills the list with `null`, which is a problem.
+
+Change it to `List.filled(length, default)` instead.
+
+## Resources
+
+*   [Null safety DartPad](https://nullsafety.dartpad.dev/)
+*   [Sound null safety](https://dart.dev/null-safety)
```
Comment (kwalrath) [inline src/null-safety/faq.md:292]: If possible, links to dart.dev should be absolute paths without a hostname.

```suggestion
*   [Null safety homepage](/null-safety)
```

File: src/null-safety/faq.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+The default 'List' constructor isn't available when null safety is enabled. #default_list_constructor
+```
+
+The default list constructor fills the list with `null`, which is a problem.
+
+Change it to `List.filled(length, default)` instead.
+
+## Resources
+
+*   [Null safety DartPad](https://nullsafety.dartpad.dev/)
```
Comment (kwalrath) [inline src/null-safety/faq.md:291]:
```suggestion
*   [DartPad with Null Safety](https://nullsafety.dartpad.dev)
```

**Review by kwalrath — COMMENTED** [review]
Noticed a few things within the text...

File: .firebaserc [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "projects": {
-    "default": "dart-dev",
+    "default": "rm-dart",
```
Comment (kwalrath) [inline .firebaserc:3]: Yikes, I don't think you meant to do this!

File: src/null-safety/faq.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
+---
+title: Null safety: frequently asked questions
+description: FAQs to help you migrate your Dart code to null safety
+short-title: FAQ (null safety)
+---
+
+This page collects some common questions we've heard about [null safety](/null-safety)
+based on the experience of migrating Google internal code.
```
Comment (kwalrath) [inline src/null-safety/faq.md:8]: 👍  Nicely worded

Comment (ryanmcginnis) [reply src/null-safety/faq.md:8]: thanks!

**Comment by ryanmcginnis** [comment]
i'm going to try again without the commit mistake. :-)

**Pull request closed without merging**